plugins/modules/packaging/os/pacman.py

```diff
@@ -505,10 +515,24 @@ def update_package_db(self):
             cmd += self.m.params["update_cache_extra_args"]
         if self.m.params["force"]:
             cmd += ["--refresh"]
+        else:
+            # Dump package database to get contents before update
+            dummy, pre_state, dummy = self.m.run_command([self.pacman_path, '--sync', '--list'], check_rc=True)
+            pre_state = sorted(pre_state.splitlines())
```

Comment:
would have been nice to be able to use `self.inventory['available_packages']` for this in order to avoid calling `pacman -Sl` 3 times, but the format of the inventory isn't the same.

Reply:
I've implemented a simple caching in 4c4442175abcae789960324484042563e7d31374 so it's only called twice when `upgrade=true` and `update_cache=true`. (The same code will make sure this is only called twice if `name` is specified and `update_cache=true`.)